packages/lit-dev-content/src/components/litdev-tutorial.ts

```diff
@@ -4,32 +4,51 @@
  * SPDX-License-Identifier: BSD-3-Clause
  */
 
-import {LitElement, html, PropertyValues} from 'lit';
+import {LitElement, html} from 'lit';
 import {property, state} from 'lit/decorators.js';
 import {unsafeHTML} from 'lit/directives/unsafe-html.js';
+import {when} from 'lit/directives/when.js';
 import {PlaygroundProject} from 'playground-elements/playground-project.js';
-import {manifest, TutorialStep} from './litdev-tutorial-manifest.js';
 import {addModsParameterToUrlIfNeeded} from '../mods.js';
 import {
   getCodeLanguagePreference,
   CODE_LANGUAGE_CHANGE,
 } from '../code-language-preference.js';
 import {solveIcon} from '../icons/solve-icon.js';
 import {resetIcon} from '../icons/reset-icon.js';
+import {catalogIcon} from '../icons/catalog-icon.js';
+import {forwardArrowIcon} from '../icons/forward-arrow-icon.js';
+import {backArrowIcon} from '../icons/back-arrow-icon.js';
 
 import '@material/mwc-icon-button';
 import './litdev-example-controls.js';
 import './litdev-playground-change-guard.js';
 import './litdev-icon-button.js';
+import {Task, TaskStatus} from '@lit-labs/task';
 
-interface ExpandedTutorialStep extends TutorialStep {
+export interface TutorialStep {
+  title: string;
+}
+
+/* TODO(e111077): find a way to make a single source of truth for this.
+ * Lives here and in /samples/tutorials/utils.cjs but samples is not in the 11ty
+ * root so they can't be imported
+ */
+export interface TutorialManifest {
+  steps: TutorialStep[];
+}
+
+interface ExpandedTutorialStep {
   idx: number;
   url: string;
   htmlSrc: string;
   projectSrcBefore: string;
   projectSrcAfter: string;
 }
 
+const samplesRoot =
+  getCodeLanguagePreference() === 'ts' ? '/samples' : '/samples/js';
```

Comment:
The language preference can change at any time when the user clicks the toggle from within the tutorial, so does this change break the ability to switch between languages without a reload?